### PR TITLE
Bugfix/sm make bcs update

### PR DESCRIPTION
**Pull request by JustinPerket**
@smahanam fixed make_bcs utilitiy to work with cmake.

Despite `bugfix` name, also adds a zero-diff feature.  **PEATMAP data processing**. Disabled by default with logical parameter process_peat=.false.  

Other changes: 

- values of pi changed to MAPL_PI
- 8 more climatological (fPAR, VISDF, NIRDF stats for CATCHCN) data files and new clsm/country_and _state_code.data file. 
- Now an extra column on soil_param.* files 
This branch produced zero-diff BCs with NLv3 (checked CF90atmos-CF90atmos and M36 SMAP grid)

This addresses https://github.com/GEOS-ESM/GEOSgcm/issues/42

### CODE REVIEW

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/mkLatLonRaster.F90 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-!   $Id$
+!   $Id:
```
Comment (JustinPerket) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/mkLatLonRaster.F90]:
```suggestion
```

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/mkMITAquaRaster.F90 [hunk 2]
```diff
@@ -1,6 +1,4 @@
-
-
-!  $Id$
+!  $Id:
```
Comment (JustinPerket) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/mkMITAquaRaster.F90]:
```suggestion
```

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/mkMOMAquaRaster.F90 [hunk 3]
```diff
@@ -1,5 +1,4 @@
-
-!   $Id$
+!   $Id:
```
Comment (JustinPerket) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/mkMOMAquaRaster.F90]:
```suggestion
```

File: GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/rasterize.H [hunk 4]
```diff
@@ -1,5 +1,5 @@
 
-!  $Id$
+!  $Id: rasterize.H,v 1.3 2011/01/19 14:00:05 msuarez Exp $
```
Comment (JustinPerket) [inline GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/rasterize.H]:
```suggestion
```

**Comment by JustinPerket** [comment]
As @smahanam points out, I commented on the issue page, and not this PR:
https://github.com/GEOS-ESM/GEOSgcm_GridComp/pull/92#issuecomment-553459018
> build fails with:
> 
> ```
> root/project/GEOSgcm/src/Components/@GEOSgcm_GridComp/GEOSagcm_GridComp/GEOSphysics_GridComp/GEOSsurface_GridComp/Shared/Raster/zip.c:4:10: fatal error: zlib.h: No such file or directory
>  #include "zlib.h"
> ```
> 
> presumably an easy fix

**Comment by mathomp4** [comment]
@JustinPerket @smahanam Interesting. Baselibs should definitely have `zlib.h` but its possible we don't handle C code as well as Fortran code. 

I'll try building myself and if that works, it points to something odd in the containers @pchakraborty uses, but I'm betting on CMake...

**Comment by mathomp4** [comment]
Okay. I might know what's happening. I need to consult @tclune and @pchakraborty on the best way to solve it. There's a cheap-and-easy way to solve and a more correct way. Hmm.

**Comment by JustinPerket** [comment]
interesting, thanks @mathomp4

**Comment by mathomp4** [comment]
@JustinPerket @smahanam I think I have a fix. I'm going to push the CMakeLists.txt on your branch because, unfortunately, the only way to test this is to have CircleCI launch a new test.

**Comment by smahanam** [comment]
I see @mathomp4 it is this automated CircleCI that has this issue. I just built this branch through GEOSldas too.

**Comment by mathomp4** [comment]
I now found a bug with gfortran being strict pushing again.

**Comment by mathomp4** [comment]
Oh dear. This next error I think means we need to use GCC 9 for the CI testing. @tclune might know more, but I think `findloc()` was a Fortran 2008 function that wasn't supported in GCC until 9.

**Comment by smahanam** [comment]
My apologies if this has caused a problem. I can work around the code as well if it is more convenient.
I can replace it with  a loop.

**Comment by mathomp4** [comment]
> My apologies if this has caused a problem. I can work around the code as well if it is more convenient.
> I can replace it with a loop.

@smahanam Let me talk with @tclune about this. `findloc()` is valid Fortran and it's only due to an old compiler this is happening. I like `findloc()` save for the fact it returns an array. Did some tests and it looks like it's as fast or faster than a brute force loop and definitely more readable.

**Comment by mathomp4** [comment]
I think I have a fix for this. It'll first require some changes go into `ESMA_cmake` that needs @sdrabenh to then update `Develop.cfg`. After that, I can then add my changes here and that should let it pass. Phew. Getting close!

**Comment by smahanam** [comment]
Thank you @mathomp4 ! As I understand we are not going backward, right? We don't have to go backward to accommodate FINDLOC, I can fix my code instead.

**Comment by mathomp4** [comment]
No, the way it will be is CMake will test if a compiler supports FINDLOC(). If not, we’ll compile a new source file, findloc.F90, and then set an ifdef to use the findloc() in there. That new file just has a native brute force loop version of findloc() that Intel should never see.

**Comment by smahanam** [comment]
@zhaobin74 @atrayano @yvikhlya 

I added Atanas, Bin Zhao and Yuri to this PR request since we all contribute to or use this Shared/Raster directory where make_bcs is run.

**Comment by mathomp4** [comment]
Okay. I've pushed a new file: `findloc.F90` that has a *very limited* version of `findloc()`, the one-d `findloc(array,value)` case. Now if an older compiler doesn't support `findloc()` it will build and use this.

NOTE: To get this functionality, you need ESMA_cmake v1.0.11. @sdrabenh changed `Develop.cfg` in the `GEOSgcm` to use this. Hopefully the CircleCI will let this build now.

**Comment by JustinPerket** [comment]
> Hopefully the CircleCI will let this build now.
looks like it does. Thanks @mathomp4 !

**Comment by smahanam** [comment]
So CircleCI  passed.
Yes Thanks so much @mathomp4

**Comment by JustinPerket** [comment]
I think this is ready for review then. Will contact @sdrabenh and @wmputman

**Comment by tclune** [comment]
Bonus points go to the person that remembers to delete the CMake test (and test program) in some future era when all of the compilers support ```findloc()``.

**Comment by mathomp4** [comment]
Per @smahanam, who discovered that my `findloc` was insanely incorrect in the sense that it always returned 0.

The odder question is *how* @smahanam found this out. Intel shouldn't have triggered this code since it supports the intrinsic `findloc()`. Odd.

**Pull request merged**